### PR TITLE
Show eReceipts for distance requests

### PR DESCRIPTION
**Pull request by neil-marcellini**
### Details

When a distance request receipt image is clicked from the transaction report the attachment modal is opened. For distance requests an eReceipt is displayed instead of the full size image of the route. The amount, merchant, waypoints list, and created date are shown.

For now, the download button will only download the route image. In the future maybe we can download the entire eReceipt view.

### Fixed Issues

$ https://github.com/Expensify/App/issues/27050
$ https://github.com/Expensify/App/issues/28289
$ https://github.com/Expensify/App/issues/28949
PROPOSAL: N/A

### Tests

Distance request with 2 waypoints

1. Click the green plus, request money, distance
2. Enter valid start and finish locations
3. Create the request
4. Open the money request report
5. Click on the distance request to open the transaction report
6. Click on the route image
7. Verify that the distance eReceipt is shown and looks good

Distance request with 12 waypoints

1. Click the green plus, request money, distance
2. Enter valid start and finish locations, plus 10 stops
3. Create the request
4. Open the money request report
5. Click on the distance request to open the transaction report
6. Click on the route image
7. Verify that the distance eReceipt is shown and looks good
8. Verify that you can scroll down to see the entire eReceipt

Also run the "action performed" from the 2nd and 3rd linked issues.

- [ ] Verify that no errors appear in the JS console

### Offline tests

Run the same tests while offline and verify you see TBD for some values
Go back online
Verify that the value fill in

### QA Steps

Same as tests
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/26260477/1bb44692-fb0e-4781-a5cf-046c3cba4e5e

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/26260477/0e691f25-0536-4a9f-89e1-61e10f42f1ea

https://github.com/Expensify/App/assets/26260477/1d85fdc1-01fe-401d-a966-a60872d76e47

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/26260477/6cc6b6e1-60d9-4daf-8f7e-95d332a616da

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/26260477/3741f9dd-9515-4748-9d43-8ace7f3aeb54

</details>

<details>
<summary>iOS</summary>

https://drive.google.com/file/d/1-9bGeQcr-sFNX_8Q5wII57MUXeT3Jzg-/view?usp=sharing
</details>

<details>
<summary>Android</summary>

https://drive.google.com/file/d/10utdOdYCBTRV_tkUkHA6cB_tJxuyt7Q6/view?usp=sharing
</details>

### CODE REVIEW

**Comment by neil-marcellini** [comment]
Here's what I have so far. It doesn't match the mock exactly but it's pretty close. May I please have an initial design review? I still need to test the receipt with many waypoints (15) and get the PR ready for final review.

image


https://github.com/Expensify/App/assets/26260477/5da17338-29ee-4592-b711-f1b33b5077d1

**Comment by shawnborton** [comment]
This feels pretty good to me, cc @Expensify/design 

The only major thing I am seeing is, should we vertically center the receipt in the available area when you view it? 

We might want to coordinate with @georgia too on how this code overlaps with eReceipts for cards.

File: src/styles/themes/default.js
Comment (shawnborton) [inline src/styles/themes/default.js:85]: Not sure how I feel about this name... maybe it's better to just have a series of helper classes like `colorGreen400` and then just call that? Otherwise textBrand doesn't feel very descriptive and I'm not sure how/where else we'd use it?

File: src/styles/themes/default.js
Comment (shawnborton) [inline src/styles/themes/default.js:86]: Same with this one, I'm not sure how I feel about this as a name.

**Comment by neil-marcellini** [comment]
Note: There's one prop type error in the test videos but I think it's reproducible on main. There's also a handful of warnings. Maybe the C+ can check.

**Comment by shawnborton** [comment]
In this view:
CleanShot 2023-10-04 at 07 24 54@2x


I find it a little weird that the receipt itself scrolls. I think the receipt should just be super long, and if anything, the attachment modal would be able to scroll to see everything. Thoughts @Expensify/design ?

**Comment by dannymcclain** [comment]
@shawnborton Yeah I agree that the receipt should just be as long as it needs to be and you should scroll inside the modal.

File: assets/images/eReceipt_background.svg
Comment (neil-marcellini) [reply assets/images/eReceipt_background.svg:1]: You can download it here 🙂 https://github.com/Expensify/App/blob/0a04398d12f0d68873a8fafd95227d08de8f5be3/assets/images/eReceipt_background.svg

**Comment by neil-marcellini** [comment]
> I find it a little weird that the receipt itself scrolls. I think the receipt should just be super long, and if anything, the attachment modal would be able to scroll to see everything. Thoughts @Expensify/design ?

I didn't get this at first, but I think what you mean is that the scroll bar should be on the modal instead of on the receipt. I'll work on fixing that soon.

**Comment by shawnborton** [comment]
That's correct. Thanks!

**Comment by neil-marcellini** [comment]
Ok, fixed! Lmk if you would like to see any other style changes.

https://github.com/Expensify/App/assets/26260477/0888feba-a629-4318-a774-d61854fb7bd5


https://github.com/Expensify/App/assets/26260477/80f0c5a3-abb1-49e6-b209-ea020972317f

**Comment by shawnborton** [comment]
Love it, much better

**Comment by mollfpr** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

File: src/components/DistanceEReceipt.js
Comment (mollfpr) [inline src/components/DistanceEReceipt.js:34]: I have a little worry about this. In my understanding, the `transaction` is optional, but the component will break if we don't pass the `transaction` param.

We can keep the `transaction` as is, and to prevent the component break, we need to check if the transaction has a receipt.

```suggestion
    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction.receipt.source, transaction.filename) : {};
```

Screenshot_1696529250

**Review by mollfpr — CHANGES_REQUESTED** [review]
Looks good and tests well. Just left few improvement.

File: src/styles/styles.js
Comment (neil-marcellini) [reply src/styles/styles.js:3257]: I discussed with @shawnborton and the rest of the design team that it's ok to use a specific color here, since this is a style that will be the same across all themes.

File: src/components/Attachments/AttachmentView/index.js
Comment (neil-marcellini) [reply src/components/Attachments/AttachmentView/index.js:58]: Oh good catch thank you.

File: src/components/DistanceEReceipt.js
Comment (neil-marcellini) [reply src/components/DistanceEReceipt.js:34]: Oh yeah good point! That looks wrong. I'll get it cleaned up soon.

**Review by mollfpr — COMMENTED** [review]
Looks good and tests well 👍

I think we can move forward, I can't find the proper step to reproduce the issue I mention. I'll do the test again after the conflict fix and approve the PR.

File: src/components/DistanceEReceipt.js
Comment (neil-marcellini) [reply src/components/DistanceEReceipt.js:94]: That's weird. I fixed it instead by removing undefined waypoints when setting up the sorted waypoints.

File: src/styles/styles.js
Comment (grgia) [reply src/styles/styles.js:3297]: Can we still define it in the themes file? importing `colors.js` into styles.js  in the past resulted in a lot of contributors using incorrect colors

Comment (grgia) [reply src/styles/styles.js:3297]: You can see how we use CONST colors for the animated page headers i.e
[SCREENS.SETTINGS.WORKSPACES]: colors.pink800,

File: src/components/Attachments/AttachmentCarousel/CarouselItem.js
Comment (grgia) [inline src/components/Attachments/AttachmentCarousel/CarouselItem.js:37]: Where is this being passed in? (Trying to understand patterns for ECard transactions eReceipts). 

I'm having trouble finding where the URL is coming from, do you save it to the receipt key in the transaction object in the BE?

**Comment by neil-marcellini** [comment]
Hey @mollfpr, I had to request a new C+ in Slack [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1697132685915229) because it has been too long since the last review.

**Comment by mananjadhav** [comment]
Reviewing the PR and the previous conversations. Thanks for resolving the conflicts @neil-marcellini.

File: src/languages/en.ts
Comment (mananjadhav) [inline src/languages/en.ts:1843]: I don't think we need this block as it already exists on L1850,=,

File: src/languages/es.ts
Comment (mananjadhav) [inline src/languages/es.ts:2328]: Same here as well, it exists  on L2334.

**Review by mananjadhav — APPROVED** [review]
Code change looks fine except for one minor object I added.

File: src/languages/en.ts
Comment (neil-marcellini) [reply src/languages/en.ts:1843]: Oh woops, thanks.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/3069065/51f2694a-a564-4b72-9d52-7246125eeab7


https://github.com/Expensify/App/assets/3069065/71069681-920c-4bb8-aca7-53e4bf1bb95f

web-distance-card

</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/3069065/6017ee77-2bba-4473-9273-1c23c302ba25

</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mananjadhav** [comment]
@neil-marcellini All done from my side.

🎀 👀 🎀 C+ reviewed.

File: src/components/Attachments/AttachmentCarousel/CarouselItem.js
Comment (grgia) [reply src/components/Attachments/AttachmentCarousel/CarouselItem.js:37]: Thank you! This was super helpful

**Review by grgia — APPROVED** [review]

File: src/components/DistanceEReceipt.js
Comment (rushatgabhane) [inline src/components/DistanceEReceipt.js:66]: Hi, we had a bug here - App crashes when tapping on the receipt thumbnail while receipt is generating

On native, the pending attachment uses a static image, and it has the source: 1 (that's a standard RN, see [this](https://stackoverflow.com/questions/52398714/react-native-requireimage-returns-number)), which is not a string so it breaks the Str.isPDF() and Str.isImage() checks.

More details: https://github.com/Expensify/App/issues/28814

**Pull request merged**
